tools/mason/MasonInit.chpl

```diff
@@ -37,44 +37,71 @@ Initialises a library project in a project directory
 */
 proc masonInit(args) throws {
   try! {
-    var name = '';
+    var dirName = '';
     var show = false;
+    var packageName = '';
+    var countArgs = 2;
     for arg in args[2..] {
-      if arg == '-h' || arg == '--help' {
-        masonInitHelp();
-        exit();
-      }
-      else if arg == '--show' {
-        show = true;
-      }
-      else {
-        name = arg;
+      countArgs += 1;
+      select (arg) {
+        when '-h' {
+          masonNewHelp();
+          exit();
+        }
+        when '--help' {
+          masonNewHelp();
+          exit();
+        }
+        when '--show' {
+          show = true;
+        }
+        when '--name' {
+          packageName = args[countArgs];
+          dirName = args[2];
+        }
+        otherwise {
+          if arg.startsWith('--name') {
```

Comment:
You could make this clearer by specifying the exact case we are checking for here (since a previous case checks for `--name`:

```suggestion
          if arg.startsWith('--name=') {
```

Reply:
Same goes for `MasonNew`